### ISSUE TITLE
POSIX Time Issue on older glibc

### ISSUE THREAD

**Issue by eezstreet** [glfw/glfw#2139]
OS and version: CentOS 6.6
Compiler version: GCC v4.8.5
Release or commit: lastest master as of writing (`da6713cd096a40a4512f468b34c189017e73f987`)
Build log:

```
[root@centos6dev build]# make
Scanning dependencies of target glfw
[  1%] Building C object src/CMakeFiles/glfw.dir/posix_time.c.o
/root/test/glfw-master/src/posix_time.c: In function ‘_glfwPlatformInitTimer’:
/root/test/glfw-master/src/posix_time.c:42:31: error: ‘CLOCK_REALTIME’ undeclared (first use in this function)
     _glfw.timer.posix.clock = CLOCK_REALTIME;
                               ^
/root/test/glfw-master/src/posix_time.c:42:31: note: each undeclared identifier is reported only once for each function it appears in
/root/test/glfw-master/src/posix_time.c:47:5: warning: implicit declaration of function ‘clock_gettime’ [-Wimplicit-function-declaration]
     if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
     ^
/root/test/glfw-master/src/posix_time.c:47:23: error: ‘CLOCK_MONOTONIC’ undeclared (first use in this function)
     if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
                       ^
make[2]: *** [src/CMakeFiles/glfw.dir/posix_time.c.o] Error 1
make[1]: *** [src/CMakeFiles/glfw.dir/all] Error 2
make: *** [all] Error 2
```

**The issue:** CLOCK_REALTIME and CLOCK_MONOTONIC might not be defined, however they may still be *available* if used in a non-standard way. This mostly affects older Linux distros.

Adding these three lines to the beginning of `src/posix_time.c` would fix the issue:

```c
#define POSIX_REQUIRED_STANDARD 199309L
#define _POSIX_C_SOURCE POSIX_REQUIRED_STANDARD
#define _POSIX_SOURCE POSIX_REQUIRED_STANDARD
```

I've added them myself and the library compiles fine.

**Potential issues:** None that I can see anyway.

**Comment by spaderthomas** [comment]
Hi, any chance this can be merged into mainline? I have a repro here, I'd be happy to provide logs or testing if needed.

**Comment by VVD** [comment]
#1988
